music_assistant/controllers/player_queues.py

```diff
@@ -1153,13 +1236,31 @@ async def _load_item(
                 queue_item.media_item.item_id,
                 queue_item.media_item.provider,
             ):
-                queue_item.media_item = library_item
+                if not isinstance(library_item, Track):
+                    self.logger.error(
+                        "Library returned wrong type: expected Track but got %s for item %s",
+                        type(library_item).__name__,
+                        queue_item.uri,
+                    )
+                # Continue with original item rather than crashing
+                else:
+                    queue_item.media_item = library_item
             elif not queue_item.media_item.image or queue_item.media_item.provider.startswith(
                 "ytmusic"
             ):
                 # Youtube Music has poor thumbs by default, so we always fetch the full item
                 # this also catches the case where they have an unavailable item in a listing
-                queue_item.media_item = await self.mass.music.get_item_by_uri(queue_item.uri)
+                fetched_item = await self.mass.music.get_item_by_uri(queue_item.uri)
+                if not isinstance(fetched_item, Track):
```

Comment:
Do we really need to guard this? Have there been occurrences where the library returns unexpected media types?